### PR TITLE
feat(processor): add sumologicsyslogprocessor

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

### CODE REVIEW

**Review by pmatyjasek-sumo — APPROVED** [review]

File: pkg/processor/sumologicsyslogprocessor/processor.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			// iterate over Logs
+			logs := ill.Logs()
+			for k := 0; k < logs.Len(); k++ {
+				var (
+					value string = syslogSource
+					ok    bool
+				)
+
+				log := logs.At(k)
+				r, _ := regexp.Compile(`^<(?P<number>\d+)>`)
```
Comment (pmalek-sumo) [inline pkg/processor/sumologicsyslogprocessor/processor.go:92]: This can be compile just once and then reused. Might be a separate PR + please check the err 😬

**Review by pmalek-sumo — APPROVED** [review]
